src/analyses/basePriv.ml

```diff
@@ -579,12 +596,25 @@ struct
     {st with cpa= cpa_local }
 
   let threadenter ask (st: BaseComponents (D).t): BaseComponents (D).t =
-    (* Copy-paste from Base make_entry *)
-    let globals = CPA.filter (fun k v -> is_global ask k) st.cpa in
-    (* let new_cpa = if !earlyglobs || ThreadFlag.is_multi ctx.ask then CPA.filter (fun k v -> is_private ctx.ask ctx.local k) globals else globals in *)
-    let new_cpa = globals in
+    (* We cannot copy over protected things, the thread may start with things privatized that are overwritten before becoming public *)
```

Comment:
Is there an example of this? How can it already privatize something if the created thread (not the creator thread!) starts with no locks?

Reply:
The comment sounds backwards then: "the thread may start with things privatized". I guess what it really should say is that the creator thread may have things privatized, but the created thread cannot have anything privatized on entry, so the privatized things of the creator must be removed.

Reply:
Yeah, it was referring to the situation before and why that was wrong which isn't terribly useful after we've fixed it.